scripts/create_joined_view.py

```diff
@@ -396,7 +397,40 @@ def main(args):
                                     assignment_fields + assignment_type_fields+ worker_fields,
                                     dispatcher_fields)
     logger.info(f"Final Item: {final_item.title}")
-    logger.info("Completed")
+    if args.create_dashboard:
+        logger.info("Creating dashboard")
+        
+        # create new webmap
+        map_item = project.dispatcher_webmap.save(
+            item_properties={"title": project.title + " Dashboard Map", "tags": [], "snippet": "Dashboard Map"})
+        new_webmap = WebMap(map_item)
+        
+        # swizzle in joined layer instead of assignments layer
+        for i, layer in enumerate(new_webmap.layers):
+            if layer["id"] == "Assignments_0":
+                new_webmap.remove_layer(layer)
+                new_webmap.add_layer(final_item)
+                new_webmap.layers[i]["id"] = "Assignments_0"
+                break
+        new_webmap.update()
+        
+        # clone dashboard with your data instead of our data
+        item = gis.content.get("af7cd356c21a4ded87d8cdd452fd8be3")
+        item_mapping = {'377b2b2014f24b0ab9b053d9b2fed113': final_item.id,
+                        'e1904f5c56484163a021155f447adf34': project.workers_item.id,
+                        'bb7d2b495ecc4ea7810b28f16ef71cba': new_webmap.item.id}
+        cloned_items = gis.content.clone_items([item], item_mapping=item_mapping, search_existing_items=False)
+        if len(cloned_items) == 0:
+            raise ValueError("Creating dashboard failed")
+            
+        # Save new name and share to group
+        logger.info("Updating title and sharing to project group")
+        new_title = project.title + " Dashboard"
+        cloned_items[0].update(item_properties={"title": new_title},
+                               thumbnail="https://www.arcgis.com/apps/opsdashboard/assets/images/no-dashboard-thumb-84c2afc9d73774823c7865b4cc776b9b.png")
```

Comment:
not sure if you want to rely on this. I'm pretty sure it's a finger-printed resource that will change when dashboard pushes an update.

Reply:
👍 removed - we don't really need to set a thumbnail